vnext/Microsoft.ReactNative/IJSValueReader.idl

```diff
@@ -16,29 +17,69 @@ namespace Microsoft.ReactNative {
     Double,
   };
 
-  // Forward only reader for JSON like streams or trees.
   [webhosthidden]
-  DOC_STRING("`IJSIValueReader` is used to read data from JavaScript in custom native modules.  It acts as a stream and supports all the types in @JSValueType.")
-  interface IJSValueReader {
-    DOC_STRING("Returns the type of the current value.")
+  DOC_STRING(
+    "Forward-only reader for JSON-like streams.\n"
+    "It is used to read data sent between native modules and the Microsoft.ReactNative.dll.\n"
+    "\n"
+    "The JSON-like streams are data structures that satisfy the [JSON specification](https://tools.ietf.org/html/rfc8259). "
+    "The data structure may have objects with name-value pairs and arrays of items. "
+    "Property values or array items can be of type `Null`, `Object`, `Array`, `String`, `Boolean`, or `Number`. "
+    "The `IJSValueReader` treats the `Number` type as `Int64` or `Double`. See @JSValueType.\n"
+    "\n"
+    "When `IJSValueReader` reads data it must walk the whole tree without skipping any items. "
+    "It means that if the current value type is `Object`, then we must call @.GetNextObjectProperty "
+    "to start reading its properties, and if the current type is `Array`, then we must call "
+    "@.GetNextArrayItem to start reading its items.\n"
+    "\n"
+    "See the @IJSValueWriter for the corresponding writer interface.\n"
+    "\n"
+    "The @IJSValueReader and @IJSValueWriter should be rarely used directly. "
+    "They are usually used to create serializer and deserializer functions for a type. "
+    "In the rest of our code we must use these functions to serialize/deserialize our values. "
```

Comment:
whose code does "our code" refer to? Can we replace it with something specific, e.g. either "app code", "React Native for Windows framework code", etc?

Reply:
I have changed the wording.